### ISSUE TITLE
AWSBedrockRuntime fails to compile

### ISSUE THREAD

**Issue by jbelkins** [awslabs/aws-sdk-swift#1159]
### Describe the bug

The `AWSBedrockRuntime` service fails to compile.

Compile errors received:
```
/Users/jbelkins/Code/Xcode/aws-sdk-swift/Sources/Services/AWSBedrockRuntime/models/Models.swift:623:39: error: value of type 'ModelStreamErrorException' has no member 'originalMessage'
        if let originalMessage = self.originalMessage {
                                 ~~~~ ^~~~~~~~~~~~~~~
/Users/jbelkins/Code/Xcode/aws-sdk-swift/Sources/Services/AWSBedrockRuntime/models/Models.swift:626:42: error: value of type 'ModelStreamErrorException' has no member 'originalStatusCode'
        if let originalStatusCode = self.originalStatusCode {
                                    ~~~~ ^~~~~~~~~~~~~~~~~~
/Users/jbelkins/Code/Xcode/aws-sdk-swift/Sources/Services/AWSBedrockRuntime/models/Models.swift:882:81: error: cannot find type 'internalServerException' in scope
                    return try decoder.decode(responseBody: message.payload) as internalServerException
                                                                                ^~~~~~~~~~~~~~~~~~~~~~~
/Users/jbelkins/Code/Xcode/aws-sdk-swift/Sources/Services/AWSBedrockRuntime/models/Models.swift:882:40: error: type 'any Error' cannot conform to 'Decodable'
                    return try decoder.decode(responseBody: message.payload) as internalServerException
                                       ^
/Users/jbelkins/Code/Xcode/aws-sdk-swift/Sources/Services/AWSBedrockRuntime/models/Models.swift:882:40: note: only concrete types such as structs, enums and classes can conform to protocols
                    return try decoder.decode(responseBody: message.payload) as internalServerException
                                       ^
/Users/jbelkins/Code/Xcode/smithy-swift/Sources/ClientRuntime/Serialization/Decoder/ResponseDecoder.swift:9:10: note: required by instance method 'decode(responseBody:)' where 'T' = 'any Error'
    func decode<T: Decodable>(responseBody: Data) throws -> T
         ^
error: fatalError
```

### Expected Behavior

AWS SDK for Swift compiles & runs with all services.

### Current Behavior

AWS SDK for Swift fails to compile with compile errors listed above.

### Reproduction Steps

Update service models to include latest
Code-generate SDK
Run `swift test`

### AWS SWIFT SDK version used

main / latest

### Compiler and Version used

Xcode 15.0

### Operating System and version

macOS 13.6

**Comment by jbelkins** [comment]
The cause of the compile failure is that aws-sdk-swift did not generate correct code for an event streaming operation defined in the `restJson1` protocol and that defines an error to be sent in the stream.

Fixes are being prepared for release:
https://github.com/smithy-lang/smithy-swift/pull/596
https://github.com/awslabs/aws-sdk-swift/pull/1160

**Comment by jbelkins** [comment]
Released in AWS SDK for Swift 0.27.0
